common/static/common/js/discussion/discussion_router.js

```diff
@@ -102,11 +108,43 @@
             };
 
             DiscussionRouter.prototype.showThread = function(forum_name, thread_id) {
+                var self, callback;
+                self = this;
                 this.thread = this.discussion.get(thread_id);
+
+                if (!this.thread){
```

Comment:
Can you add a comment explaining what circumstances could cause `this.thread` to be falsy?